### ISSUE TITLE
cg_playerName should be latin letters only

### ISSUE THREAD

**Issue by HuskyGFX** [yvt/openspades#422]
I am dealing with at least one cheater per session recently and cannot kick many of them since they abuse special characters including "#" and numbers which prevent them from getting votekicked. Therefore I think that cg_playerName should allow latin letters (A-Z) as input only.

**Comment by noway421** [comment]
In pyspades/pysnip commands that take player names as arguments usually allow to specify player id instead. Try using something like `/kick #12`

**Comment by Mduro** [comment]
The way Aloha does it is that any player whose name has the character `#` are not allowed to enter the server. Regular players can't votekick the player (i.e. `/votekick #abcd` doesn't work), but staff can still `/kick` or `/ban` as usual, like noway421 said.

I would suggest that you autokick players whose name has the character `#`. Even if a fix for this is released, they can still use the old version of OpenSpades to maliciously avoid being votekicked.

**Comment by HuskyGFX** [comment]
@noway421 I am aware of that command but the system fails to kick properly if you name yourself "#0" for example to avoid getting kicked if somebody else has #0 as id. And somehow players cannot be kicked either that just have "#" as prefix: A guy named "#SWAG" cannot be kicked since they do not exist to the system eventhough you'd enter "/votekick #9" which was their id.

**Comment by feikname** [comment]
Most cheats (like WTFAOS) are for classic and not for OpenSpades anyway. This is a server protocol issue and the client is only following it. In my opinion, this should remain as is, and server side fix for everyone should be provided instead.

**Comment by theunknownxy** [comment]
The problem seems to be that there is no interest in fixing this bug. The bug is basically that some scripts like ratio are broken and throw an exception if the username begins with an "#", which will abort the votekick procedure.
I wrote a [workaround](https://gist.github.com/theunknownxy/fb6c63383c54f8a5a4a3) for this behaviour some time ago and I think I posted it on the pyspades/pysnip IRC channel. 
Overall I think there haven't been any big new developments on OS/AoS Classic(pyspades/pysnip/iceball, too) for quite some time and there seems to be not enough interest into reviving the game. On the other hand I'm not involved in the community at the moment and haven't played OS for a few months.


TL;DR: It's a bug in the server and the community is not really active.

**Comment by dataf3l** [comment]
today I encountered this problem
